core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1731,7 +1730,7 @@ public void testNoTrailingLocationSlash() {
 
   private String createManifestListWithManifestFile(
       long snapshotId, Long parentSnapshotId, String manifestFile) throws IOException {
-    File manifestList = temp.newFile("manifests" + UUID.randomUUID());
+    File manifestList = java.nio.file.Files.createTempDirectory(temp, "junit").toFile();
```

Comment:
same as above, shouldn't this create a temp file with the given name rather than a temp folder?

Reply:
yes it should be a temp file. Will fix this part.